samples/common/mcumgr_bt_ota_dfu/Kconfig

```diff
@@ -4,39 +4,58 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-DT_CHOSEN_EXT_FLASH:= nordic,pm-ext-flash
-
 config NCS_SAMPLE_MCUMGR_BT_OTA_DFU
 	bool "MCUmgr OTA DFU over Bluetooth"
+	select CRC
+	select FLASH
+	select FLASH_MAP
 	select MCUMGR
+	select MCUMGR_TRANSPORT_BT
 	select NET_BUF
 	select ZCBOR
-	select CRC
-	select MCUMGR_TRANSPORT_BT
-	imply MCUMGR_TRANSPORT_BT_CONN_PARAM_CONTROL
-	imply IMG_MANAGER
-	imply STREAM_FLASH
-	imply FLASH_MAP
-	imply FLASH
-	imply MCUMGR_GRP_IMG
 	imply MCUMGR_GRP_OS
 	imply MCUMGR_GRP_OS_BOOTLOADER_INFO
+	imply MCUMGR_TRANSPORT_BT_CONN_PARAM_CONTROL
 	imply MCUMGR_TRANSPORT_BT_REASSEMBLY
 	imply NCS_SAMPLE_MCUMGR_BT_OTA_DFU_SPEEDUP
 	depends on BT_PERIPHERAL
-	depends on BOOTLOADER_MCUBOOT
+	depends on BOOTLOADER_MCUBOOT || SUIT
 	help
 	  Enable this option to include MCUmgr in the build which will enable
 	  the Bluetooth transport and image management group, which allows
 	  for firmware over the air update support of the device.
 
 if NCS_SAMPLE_MCUMGR_BT_OTA_DFU
 
-config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
-	bool
-	default y if MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP
-	default y if MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT
-	imply MCUMGR_GRP_IMG_REJECT_DIRECT_XIP_MISMATCHED_SLOT
+choice NCS_SAMPLE_MCUMGR_BT_OTA_DFU_BACKEND
+	prompt "Choose DFU backend"
+	help
+	  The choice option is autoselected based on their mutually exclusive
+	  dependencies.
+
+config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_BACKEND_MCUBOOT
+	bool "Use MCUboot as DFU backend"
+	select IMG_MANAGER
+	select MCUMGR_GRP_IMG
+	select STREAM_FLASH
+	imply MCUMGR_GRP_IMG_REJECT_DIRECT_XIP_MISMATCHED_SLOT if \
+	      MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP || MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT
+	depends on BOOTLOADER_MCUBOOT
+	help
+	  This option enables the MCUboot bootloader as the backend for the MCUmgr DFU module.
+
+config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_BACKEND_SUIT
+	bool "Use SUIT as DFU backend"
+	select MCUMGR_SMP_LEGACY_RC_BEHAVIOUR
+	select MGMT_SUITFU
+	select MGMT_SUITFU_GRP_SUIT
+	select SSF_SUIT_SERVICE_ENABLED
+	select ZCBOR_CANONICAL
+	depends on SUIT
+	help
+	  This option enables the SUIT as the backend for the MCUmgr DFU module.
```

Comment:
as above, there should be no selection on there, it should be done automatically

Reply:
What do you suggest? The intention of this selection is to be automatic. However, the Kconfig choice type cannot be defined without the prompt.

The `depends on` syntax around the Kconfig choice is set up in such a way that only one variant is possible. I decided to use choice syntax to make it impossible to select more than one backend in the given build configuration.

Reply:
Remove the choice completely and prompt, have 2 Kconfigs and have them default to on if the requirements are met and `NCS_SAMPLE_MCUMGR_BT_OTA_DFU` is enabled, that way it is completely automatic

Reply:
Okay, I can do that. However, if someone misconfigures the dependencies for the 3rd backend, we may end up in a situation where more than one backend is selected. The Kconfig structurally prevents this from happening. However, I am happy to align my change with your suggestion

Reply:
The bootloader is determined by the board and enforced by sysbuild, so e.g. configuring nrf52840dk will either have MCUboot enabled or disabled, configuring nrf54h20 will have SUIT enabled. If a third is added in the future, I assume it will be sysbuild allowing the selection (if it can be changed) and enforcing it